src/useObserver.ts

```diff
@@ -4,15 +4,31 @@ import { printDebugValue } from "./printDebugValue"
 import { isUsingStaticRendering } from "./staticRendering"
 import { useForceUpdate, useUnmount } from "./utils"
 
-export function useObserver<T>(fn: () => T, baseComponentName = "observed"): T {
+export type ForceUpdateHook = () => () => void
+
+export function useObserver<T>(
+    fn: () => T,
+    baseComponentName?: string,
+    customForceUpdateHook?: ForceUpdateHook
+): T {
     if (isUsingStaticRendering()) {
         return fn()
     }
-    const forceUpdate = useForceUpdate()
+
+    const wantedForceUpdateHook = customForceUpdateHook || useForceUpdate
+    if (process.env.NODE_ENV !== "production") {
+        const forceUpdateRef = useRef<ForceUpdateHook>(wantedForceUpdateHook)
+        if (forceUpdateRef.current !== wantedForceUpdateHook) {
+            throw new Error(
+                "a custom force update hook cannot be switched to another one once used"
```

Comment:
I don't like this. It's an extra check for everyone just because there is a potential for a custom update hook. Is this an actual problem? You haven't captured it in the tests. I would say it's more about the rules of hooks. If someone would be that silly and change called hooks conditionally, they should burn in their own hell :)

Reply:
agreed, removed